h2o-algos/src/main/java/hex/tree/DHistogram.java

```diff
@@ -492,8 +494,13 @@ void updateHisto(double[] ws, double resp[], double[] cs, double[] ys, double[]
       _vals[_vals_dim*b + 1] += wy;
       _vals[_vals_dim*b + 2] += wyy;
       if (_vals_dim >= 5 && !Double.isNaN(resp[k])) { // FIXME (PUBDEV-7553): This needs to be applied even with monotone constraints disabled
-        _vals[_vals_dim * b + 3] += weight * (_pred1 - y) * (_pred1 - y);
-        _vals[_vals_dim * b + 4] += weight * (_pred2 - y) * (_pred2 - y);
+        if (_dist._family.equals(DistributionFamily.quantile)) {
+          _vals[_vals_dim * b + 3] += _dist.deviance(weight, y, _pred1);
```

Comment:
it would be worthy to extract _vals_dim * b to local variable since it is used 9 times in this single method